src/main/scala/org/scalacheck/magnolia/instances.scala

```diff
@@ -43,5 +45,29 @@ package object magnolia {
     override def map[A, B](from: Gen[A])(fn: A => B): Gen[B] = from.map(fn)
   }
 
-  implicit def gen[T]: Arbitrary[T] = macro Magnolia.gen[T]
+  def arbitrary[T]: Arbitrary[T] = macro Magnolia.gen[T]
+}
+
+object MagnoliaCogen {
+  type Typeclass[T] = Cogen[T]
+
+  def combine[T](caseClass: CaseClass[Cogen, T]): Cogen[T] = Cogen { (seed, t) =>
+    caseClass.parameters.foldLeft(seed) { (seed, param) =>
+      param.typeclass.perturb(seed, param.dereference(t))
+    }
+  }
+
+  def dispatch[T](sealedTrait: SealedTrait[Cogen, T]): Cogen[T] = Cogen { (seed, t) =>
+    sealedTrait.dispatch(t) { subtype =>
+      subtype.typeclass.perturb(seed, subtype.cast(t))
+    }
+  }
+
+  def cogen[T]: Cogen[T] = macro Magnolia.gen[T]
+}
+
+//this would end up in the package object
+object Demo {
+  // implicit def cogenInstance[T] = macro MagnoliaCogen.cogen[T]
```

Comment:
These lines basically need to be:
```
implicit def cogenInstance[T] = MagnoliaCogen.cogen[T]
implicit def arbitraryInstance[T] = MagnoliaArbitrary.arbitrary[T]
```

Reply:
I think I tried without `macro` and it still broke... but I'll give it another try.